lumicks/pylake/kymotracker/detail/gaussian_mle.py

```diff
@@ -146,38 +159,50 @@ def gaussian_mle_1d(
     if fixed_background is not None and fixed_background <= 0:
         raise ValueError("Fixed background should be larger than zero.")
 
-    initial_position = x[np.argmax(photon_count)] if initial_position is None else initial_position
-    initial_position = np.atleast_1d(initial_position)
+    initial_position = np.atleast_1d(
+        x[np.argmax(photon_count)] if initial_position is None else initial_position
+    )
 
+    num_peaks = initial_position.size
     expectation_fun = partial(
         peak_expectation_1d,
         x,
         fixed_background=fixed_background,
         pixel_size=pixel_size,
-        n_components=initial_position.size,
+        num_peaks=num_peaks,
     )
     derivatives_fun = partial(
         peak_expectation_1d_derivatives,
         x,
         fixed_background=fixed_background,
         pixel_size=pixel_size,
-        n_components=initial_position.size,
+        num_peaks=num_peaks,
     )
 
     amp_estimate = np.max(photon_count) / pixel_size * np.sqrt(2 * np.pi * initial_sigma ** 2)
-    initial_guess, bounds = [], []
-    for init_pos in initial_position:
-        initial_guess.extend([amp_estimate / initial_position.size, init_pos, initial_sigma])
-        bounds.extend([(0.01, None), (np.min(x), np.max(x)), (pixel_size, 10 * pixel_size)])
+
+    num_pars = num_peaks * 3 + (0 if fixed_background else 1)
+    initial_guess = np.empty(num_pars)
+    initial_guess[:num_peaks] = amp_estimate / initial_position.size
+    initial_guess[num_peaks : 2 * num_peaks] = initial_position
+    initial_guess[2 * num_peaks :] = initial_sigma
+
```

Comment:
maybe we could add a small comment above all this to just explicitly state the order of parameters `# amplitudes, centers, widths, background offset`. I think, for these long mixed parameter lists, it just helps to quickly be able to keep track of things without needing to decipher the code everytime you look at it

Reply:
Yeah, that's fair.